gitlab/client.py

```diff
@@ -764,7 +764,10 @@ def http_request(
                 return result
 
             if (429 == result.status_code and obey_rate_limit) or (
-                result.status_code in RETRYABLE_TRANSIENT_ERROR_CODES
+                (
+                    result.status_code in RETRYABLE_TRANSIENT_ERROR_CODES
+                    or (result.status_code == 409 and "Resource lock" in result.reason)
+                )
```

Comment:
Hehe, it seems like pylint doesn't like the chunky if condition - let's try what I suggested in https://github.com/python-gitlab/python-gitlab/issues/2325#issuecomment-1282310403.

Reply:
Done in https://github.com/python-gitlab/python-gitlab/pull/2326/commits/25446cf3297453ef58abc66e701813475423fe59.